File: demo/src/main/java/com/lcm/app/data/entity/HttpBaseResult.java
package com.lcm.app.data.entity;

/**
 * Created by hunji on 2016/9/1.
 * desc:
 */
public class HttpBaseResult<T> {

    /**
     * server_time : 1477284985
     * success : true
     * message :
     * data :
     */

    private int server_time;
    private boolean success;
    private String message;
    private T data;
    private String error_code;

    public String getError_code() {
        return error_code;
    }

    public void setError_code(String error_code) {
        this.error_code = error_code;
    }

    public int getServer_time() {
        return server_time;
    }

    public void setServer_time(int server_time) {
        this.server_time = server_time;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: demo/src/main/java/com/lcm/app/data/entity/HttpBaseResult.kt
package com.lcm.app.data.entity

/**
 * ****************************************************************
 * Author: LCM
 * Date: 2018/5/4 16:17
 * Desc:
 * *****************************************************************
 */
class HttpBaseResult<T> {

    private var error:Boolean=false
    var results:T?=null

    override fun toString(): String {
        return "HttpBaseResult(error=$error, results=$results)"
    }

    fun isError():Boolean{
        return error
    }

    fun setError(error: Boolean){
        this.error = error
    }


}